src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt

```diff
@@ -110,38 +114,17 @@ abstract class QueueIntegrationTest : IntegrationTest() {
     return count
   }
 
+  private val untilMessagesGetOnTheQueueInTheFirstPlace = Duration.ofSeconds(1)
+
   fun indexPrisoners() {
     webTestClient.put().uri("/prisoner-index/build-index")
       .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_INDEX")))
       .exchange()
       .expectStatus().isOk
 
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089EY") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089EZ") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089FA") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089FB") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089FC") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7089FX") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AA") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AB") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AC") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AD") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AE") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090AF") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BA") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BB") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BC") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BD") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BE") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A7090BF") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A9999AA") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A9999AB") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A9999RA") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A9999RB") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A9999RC") } matches { it == 1 }
-    await untilCallTo { prisonRequestCountFor("/api/offenders/A1090AA") } matches { it == 1 }
-
-    await untilCallTo { getNumberOfMessagesCurrentlyOnIndexQueue() } matches { it == 0 }
+    await
+      .atLeast(untilMessagesGetOnTheQueueInTheFirstPlace)
+      .atMost(Duration.ofSeconds(60)) untilCallTo { getNumberOfMessagesCurrentlyOnIndexQueue() } matches { it == 0 }
```

Comment:
Generalise indexPrisoners() so it can be used with different prisoner lists. Only the getNumberOfMessagesCurrentlyOnIndexQueue is needed